### ISSUE TITLE
Can't train, but no error

### ISSUE THREAD

**Issue by newoneincntk** [microsoft/CNTK#3820]
Hello, any guys here?
I use PIP to install CNTK on WIN10 1909/1903. But when I run a script, although there is no error message, the training process does not start.
My PC is AMD 3950x + 2080TI.

**Comment by delzac** [comment]
Did you cntk-gpu?  Have installed CUDA already?

You follow instructions here:
[https://docs.microsoft.com/en-us/cognitive-toolkit/setup-windows-python?tabs=cntkpy26](https://docs.microsoft.com/en-us/cognitive-toolkit/setup-windows-python?tabs=cntkpy26)

**Comment by newoneincntk** [comment]
@delzac First, I want to thank you for your great job in cntkx. it is the only hope for those still using cntk. I use cntk-gpu, and install the  up-to-date graphics drivers, but not intall the CUDA toolkit. I tested the cntk, found it can test using a trained model, but cannot train.

**Comment by delzac** [comment]
When you train the model, is cntk using cpu or gpu? cntk will tell you in the console.

example:
`Selected GPU[0] GeForce GTX 1060 with Max-Q Design as the process wide default device.`

**Comment by newoneincntk** [comment]
@delzac Yes, it correctly told me this information. I tried some examples that I was able to successfully train before in my old pc, but it cannot start training in my new pc.

**Comment by delzac** [comment]
So are you using gpu or cpu?

**Comment by newoneincntk** [comment]
@delzac GPU 2080TI

**Comment by delzac** [comment]
In task manager does it show the gpu being utilised?

Also, when you said that you can train in the old PC and not on the new PC, did the training script or model change in anyway?

**Comment by delzac** [comment]
Did the console show:
`Selected GPU[0] GeForce RTX 2080TI as the process wide default device.`

When cntk runs? This is important

**Comment by newoneincntk** [comment]
@delzac Yes, it shows in the console.

**Comment by delzac** [comment]
I have in the past tried to build a model wrongly and resulting in cntk being stuck in model compilation step forever. Where is the process being held up?

**Comment by newoneincntk** [comment]
It can tell me the parameters number and tensor to train. Seems it is stuck in the train process. @delzac

**Comment by delzac** [comment]
Seems like its not model compilation. I'm also on windows 10, AMD cpu but on 1080ti.

How do you train the model? `trainer.train_minibatch` or something else?

Did you use the 2.7.post1 meaning `cntk_gpu-2.7.post1-cp36-cp36m-win_amd64.whl` for the install?

**Comment by newoneincntk** [comment]
This is one of my test cold from cntk.

from __future__ import print_function
import numpy as np
import cntk as C
from cntk.learners import sgd
from cntk.logging import ProgressPrinter
from cntk.layers import Dense, Sequential
from cntk.device import try_set_default_device, gpu, cpu
#print('here')
def generate_random_data(sample_size, feature_dim, num_classes):
     # Create synthetic data using NumPy.
     Y = np.random.randint(size=(sample_size, 1), low=0, high=num_classes)

     # Make sure that the data is separable
     X = (np.random.randn(sample_size, feature_dim) + 3) * (Y + 1)
     X = X.astype(np.float32)
     # converting class 0 into the vector "1 0 0",
     # class 1 into vector "0 1 0", ...
     class_ind = [Y == class_number for class_number in range(num_classes)]
     Y = np.asarray(np.hstack(class_ind), dtype=np.float32)
     return X, Y

def ffnet():
    inputs = 2
    outputs = 2
    layers = 2
    hidden_dimension = 50

    # input variables denoting the features and label data
    features = C.input_variable((inputs), np.float32)
    label = C.input_variable((outputs), np.float32)

    # Instantiate the feedforward classification model
    my_model = Sequential ([
                    Dense(hidden_dimension, activation=C.sigmoid),
                    Dense(outputs)])
    z = my_model(features)

    ce = C.cross_entropy_with_softmax(z, label)
    pe = C.classification_error(z, label)
#    print('here')
    # Instantiate the trainer object to drive the model training
    lr_per_minibatch = C.learning_parameter_schedule(0.125)
    progress_printer = ProgressPrinter(0)
#    print('here')
    C.logging.log_number_of_parameters(z) ; print()
    trainer = C.Trainer(z, (ce, pe), [sgd(z.parameters, lr=lr_per_minibatch)], [progress_printer])

    # Get minibatches of training data and perform model training
    minibatch_size = 25
    num_minibatches_to_train = 1024

    aggregate_loss = 0.0
    print('here')
    for i in range(num_minibatches_to_train):
        train_features, labels = generate_random_data(minibatch_size, inputs, outputs)
        # Specify the mapping of input variables in the model to actual minibatch data to be trained with
        trainer.train_minibatch({features : train_features, label : labels})
        sample_count = trainer.previous_minibatch_sample_count
        aggregate_loss += trainer.previous_minibatch_loss_average * sample_count

    last_avg_error = aggregate_loss / trainer.total_number_of_samples_seen

    test_features, test_labels = generate_random_data(minibatch_size, inputs, outputs)
    avg_error = trainer.test_minibatch({features : test_features, label : test_labels})
    print(' error rate on an unseen minibatch: {}'.format(avg_error))
    return last_avg_error, avg_error

np.random.seed(98052)
ffnet()

**Comment by newoneincntk** [comment]
This  is the result
Selected GPU[0] GeForce RTX 2080 Ti as the process wide default device.
-------------------------------------------------------------------
Build info:

                Built time: Apr 23 2019 21:50:08
                Last modified date: Tue Apr 23 17:37:55 2019
                Build type: Release
                Build target: GPU
                With ASGD: yes
                Math lib: mkl
                CUDA version: 10.0.0
                CUDNN version: 7.3.1
                Build Branch: HEAD
                Build SHA1: ae9c9c7c5f9e6072cc9c94c254f816dbdc1c5be6 (modified)
                MPI distribution: Microsoft MPI
                MPI version: 7.0.12437.6
-------------------------------------------------------------------
 average      since    average      since      examples
    loss       last     metric       last
 ------------------------------------------------------
Training 252 parameters in 4 parameter tensors.
@delzac

**Comment by newoneincntk** [comment]
`from __future__ import print_function
import numpy as np
import cntk as C
from cntk.learners import sgd
from cntk.logging import ProgressPrinter
from cntk.layers import Dense, Sequential
from cntk.device import try_set_default_device, gpu, cpu
#print('here')
def generate_random_data(sample_size, feature_dim, num_classes):
     # Create synthetic data using NumPy.
     Y = np.random.randint(size=(sample_size, 1), low=0, high=num_classes)

     # Make sure that the data is separable
     X = (np.random.randn(sample_size, feature_dim) + 3) * (Y + 1)
     X = X.astype(np.float32)
     # converting class 0 into the vector "1 0 0",
     # class 1 into vector "0 1 0", ...
     class_ind = [Y == class_number for class_number in range(num_classes)]
     Y = np.asarray(np.hstack(class_ind), dtype=np.float32)
     return X, Y

def ffnet():
    inputs = 2
    outputs = 2
    layers = 2
    hidden_dimension = 50

    # input variables denoting the features and label data
    features = C.input_variable((inputs), np.float32)
    label = C.input_variable((outputs), np.float32)

    # Instantiate the feedforward classification model
    my_model = Sequential ([
                    Dense(hidden_dimension, activation=C.sigmoid),
                    Dense(outputs)])
    z = my_model(features)

    ce = C.cross_entropy_with_softmax(z, label)
    pe = C.classification_error(z, label)
#    print('here')
    # Instantiate the trainer object to drive the model training
    lr_per_minibatch = C.learning_parameter_schedule(0.125)
    progress_printer = ProgressPrinter(0)
#    print('here')
    C.logging.log_number_of_parameters(z) ; print()
    trainer = C.Trainer(z, (ce, pe), [sgd(z.parameters, lr=lr_per_minibatch)], [progress_printer])

    # Get minibatches of training data and perform model training
    minibatch_size = 25
    num_minibatches_to_train = 1024

    aggregate_loss = 0.0
    print('here')
    for i in range(num_minibatches_to_train):
        train_features, labels = generate_random_data(minibatch_size, inputs, outputs)
        # Specify the mapping of input variables in the model to actual minibatch data to be trained with
        trainer.train_minibatch({features : train_features, label : labels})
        sample_count = trainer.previous_minibatch_sample_count
        aggregate_loss += trainer.previous_minibatch_loss_average * sample_count

    last_avg_error = aggregate_loss / trainer.total_number_of_samples_seen

    test_features, test_labels = generate_random_data(minibatch_size, inputs, outputs)
    avg_error = trainer.test_minibatch({features : test_features, label : test_labels})
    print(' error rate on an unseen minibatch: {}'.format(avg_error))
    return last_avg_error, avg_error

np.random.seed(98052)
ffnet()`

**Comment by delzac** [comment]
I tested your code, it works for me.

Perhaps reinstall everything. Can you try install CUDA 10
